### PR TITLE
Replace react-chrome-redux with webext-redux

### PR DESCRIPTION
**Pull request by cezaraugusto**
Fix https://github.com/brave/brave-browser/issues/6777

Replacement was made since per `react-chrome-redux` readme in npm, the package was deprecated in favor of `webext-redux`.

See https://www.npmjs.com/package/react-chrome-redux for info

## Submitter Checklist:

- [ ] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by petemill** [comment]
CI reports ` Version mismatch between brave-browser (ca-6777/0.74.17) and brave-core (ca-6777/0.74.13) in package.json`

Since this isn't blocking a bug, perhaps let's get a clean CI build, at least for the web-ui builds.

**Comment by petemill** [comment]
I pushed a rebase

File: third_party/npm-react-chrome-redux/README.chromium [hunk 1]
```diff
@@ -1,4 +1,4 @@
 Name: React Chrome Redux
```
Comment (fmarier) [inline third_party/npm-react-chrome-redux/README.chromium]: You could also update this line to the new name: WebExt Redux

**Comment by bsclifton** [comment]
Restarted CI, as there were several problems here

**Comment by cezaraugusto** [comment]
rebased since CI was complaining about it

**Comment by cezaraugusto** [comment]
merging as-is since feedback addressed, PR is approved and build failure is not related to this PR and so were previous attempts I did. thanks all

**Pull request merged**